docs/csharp/programming-guide/classes-and-structs/object-and-collection-initializers.md

```diff
@@ -103,6 +103,34 @@ The preceding sample generates code that calls the <xref:System.Collections.Gene
 
 This initializer example calls <xref:System.Collections.Generic.Dictionary%602.Add(%600,%601)> to add the three items into the dictionary. These two different ways to initialize associative collections have slightly different behavior because of the method calls the compiler generates. Both variants work with the `Dictionary` class. Other types may only support one or the other based on their public API.
 
+## Object Initializers with collection read-only property initialization
+
+Some classes may have collection properties where the property is read-only, like the `Cats` property of `CatOwner` in the following case:
+
+[!code-csharp[ObjectInitializer1](../../../../samples/snippets/csharp/programming-guide/classes-and-structs/object-collection-initializers/BasicObjectInitializers.cs#CatOwnerDeclaration)]
+
+You will not be able to use collection initializer syntax discussed so far since the property cannot be assigned a new list:
+
+```csharp
+CatOwner owner = new CatOwner
+{
+    Cats = new List<Cat>
+    {
+        new Cat{ Name = "Sylvester", Age=8 },
+        new Cat{ Name = "Whiskers", Age=2 },
+        new Cat{ Name = "Sasha", Age=14 }
+    }
+};
```

Comment:
```suggestion
var owner = new CatOwner
{
    Cats = new List<Cat>
    {
        new Cat { Name = "Sylvester", Age = 8 },
        new Cat { Name = "Whiskers", Age = 2 },
        new Cat { Name = "Sasha", Age = 14 }
    }
};
```

Reply:
While I agree with all your suggestions, I deliberately used explicit typing instead of `var` and the poor formatting of the object initializers to be consistent with the rest of the examples on that page. I feel these proposed changes should be a separate PR that refreshes all examples on the page and which is a broader change to the entire document as opposed to just a new section being added here.